backends/ebpf/runtime/ebpf_runtime.c

```diff
@@ -14,89 +14,192 @@ See the License for the specific language governing permissions and
-            "feeds the individual packets into a filter function, "
+    fprintf(stderr, "This program expects a pcap file pattern, "
+            "extracts all the packets out of the matched files"
+            "in the order given by the packet time,"
+            "then feeds the individual packets into a filter function, "
             "and returns the output.\n");
-    fprintf(stderr, "Usage: %s [-d] -f file.pcap\n", name);
+    fprintf(stderr, "Usage: %s [-d] -f file.pcap -n num_pcaps\n", name);
     fprintf(stderr, "Options:\n");
     fprintf(stderr, "\t-d: Turn on debug messages\n");
     fprintf(stderr, "\t-f: The input pcap file\n");
+    fprintf(stderr, "\t-n: Specifies the number of input interfaces\n");
     exit(EXIT_FAILURE);
 }
-
-static int process_packets(pcap_t *in_handle, pcap_dumper_t *out_handle) {
-    struct pcap_pkthdr *pcap_hdr;
-    const unsigned char *packet;
-    int ret;
-    while ((ret = pcap_next_ex(in_handle, &pcap_hdr, &packet)) == 1) {
-        /* Parse each packet in the file and check the result */
+/**
+ * @brief Feed a list packets into an eBPF program.
+ * @details This is a mock function emulating the behavior of a running
+ * eBPF program. It takes a list of input packets and iteratively parses them
+ * using the given imported ebpf_filter function. The output defines whether
+ * or not the packet is "dropped." If the packet is not dropped, its content is
+ * copied and appended to an output packet list.
+ *
+ * @param pkt_list A list of input packets running through the filter.
+ * @return The list of packets "surviving" the filter function
+ */
+static pcap_list_t *feed_packets(pcap_list_t *pkt_list) {
+    pcap_list_t *output_pkts = allocate_pkt_list();
+    uint64_t list_len = get_pkt_list_length(pkt_list);
+    for (uint64_t i = 0; i < list_len; i++) {
+        /* Parse each packet in the list and check the result */
         struct sk_buff skb;
-        skb.data = (void *) packet;
-        skb.len = pcap_hdr->len;
+        pcap_pkt *input_pkt = get_packet(pkt_list, i);
+        skb.data = (void *) input_pkt->data;
+        skb.len = input_pkt->pcap_hdr.len;
         int result = ebpf_filter(&skb);
-        if (result)
-            pcap_dump((unsigned char *) out_handle, pcap_hdr, packet);
+        if (result != 0) {
+            /* We copy the entire content to emulate an outgoing packet */
+            pcap_pkt *out_pkt = copy_pkt(input_pkt);
+            output_pkts = append_packet(output_pkts, out_pkt);
+        }
+        if (debug)
+            printf("Result of the eBPF parsing is: %d\n", result);
+    }
+    return output_pkts;
+}
+
+static void write_pkts_to_pcaps(const char *pcap_base, pcap_list_array_t *output_array) {
+    uint16_t arr_len = get_list_array_length(output_array);
+    for (uint16_t i = 0; i < arr_len; i++) {
+        char pcap_out_name[FILE_NAME_MAX];
+        if (snprintf(pcap_out_name, FILE_NAME_MAX, "%s%d%s", pcap_base, i, PCAPOUT) >=FILE_NAME_MAX) {
```

Comment:
You should avoid using fixed-size buffers. The right solution is to measure the string and allocate a buffer of the right size. You should still be defensive about snprintf in case you got it wrong.